### PR TITLE
Use GenerateFileFromTemplate task from arcade instead of repo copy

### PR DESCRIPTION
**Pull request by akoeplinger**
They were both based on some aspnetcore build tooling, the only difference in the arcade one is that is uses key=value strings for the replacement properties instead of msbuild items.

Also fix an issue in workloads-testing.targets where we didn't use a separate MSBuild evaluation for the Restore task which meant that the build tasks weren't loaded in the Pack task.

### CODE REVIEW

File: eng/Tools.props [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.DotNet.Build.Tasks.Packaging" Version="$(MicrosoftDotNetBuildTasksPackagingVersion)" />
+    <PackageReference Include="Microsoft.DotNet.Build.Tasks.Templating" Version="$(MicrosoftDotNetBuildTasksTemplatingVersion)" />
```
Comment (ViktorHofer) [inline eng/Tools.props:10]: Why is this required?

Comment (akoeplinger) [reply eng/Tools.props:10]: The .pkgproj didn't work otherwise.

**Review by ViktorHofer — APPROVED** [review]
